wandb/sdk/wandb_run.py

```diff
@@ -147,14 +147,33 @@ class RunStatusChecker(object):
     For now, we just use this to figure out if the user has requested a stop.
     """
 
-    def __init__(self, interface: BackendSender, polling_interval: int = 15) -> None:
+    def __init__(self, interface: BackendSender, stop_polling_interval: int = 15, retry_polling_interval: int = 1) -> None:
         self._interface = interface
-        self._polling_interval = polling_interval
+        self._stop_polling_interval = stop_polling_interval
+        self._retry_polling_interval = retry_polling_interval
 
         self._join_event = threading.Event()
-        self._thread = threading.Thread(target=self.check_status)
-        self._thread.daemon = True
-        self._thread.start()
+        self._stop_thread = threading.Thread(target=self.check_status)
+        self._stop_thread.daemon = True
+        self._stop_thread.start()
+
+        self._retry_thread = threading.Thread(target=self.check_retries)
```

Comment:
i just added this stuff to RunStatusChecker, seems like it makes sense to have it all in one object though maybe needs a rename

Reply:
can we just rename _stop_thread to _status_thread and do both stop checking and retry checking in a single thread

Reply:
chatted with @stephchen, there are good reasons not to combine (status checker is doing a graphql request),  i agree this shouldnt be combined,  but since they are different they probably shouldnt share a message.  

Maybe your thread can be sync_status_thread, and the message called something like that SyncStatus,  or TransferStatus.   Having a thread that is getting this info is useful, and it is used for retry status in this implementation but it can easily pick up other info about how much has been synced or alert the user when the network sending threads are getting slow. etc